### PR TITLE
Add structure for user documentation

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #326

### CODE REVIEW

**Comment by neithernut** [comment]
Sections missing:
- [X] Command invocation
- [X] Files (socket etc)
  - [X] Socket(s)
  - [X] Config file (search order)
- [X] Environment

**Comment by matthiasbeyer** [comment]
I added the environment and the file files, the latter in one file by now.

I would include the command invocation in the protocol part / API description.

**Comment by neithernut** [comment]
By "command invocation" I do **not** mean ivoking API commands, but invoking `waysome` itself.

**Comment by matthiasbeyer** [comment]
Ah, okay. Is it necessary to have a dedicated section for this? Could it go into the "description" part of the documentation?

**Comment by neithernut** [comment]
Nope, the conventional manpage format requires a section about the invocation.

**Comment by matthiasbeyer** [comment]
Ok. I've added it.

File: doc/user/waysome.tex [hunk 1]
```diff
@@ -18,7 +18,23 @@
...
+% Import the title page
+\input{src/invocation}
+
+% Import the description of waysome
+\input{src/description}
+
+% Import the description of waysome
+\input{src/environment}
+
+% Import the description of waysome
```
Comment (neithernut) [inline doc/user/waysome.tex]: C&P

Comment (matthiasbeyer) [inline doc/user/waysome.tex]: Damn.

**Comment by neithernut** [comment]
Mh... I read on latex2man again.
We should have the following manpage-sections:
- Synopsis
- Description
- Options
- Files
- See Also (refering to the more detailed PDF version or website)
- Author

Those would also be `\section`s in the manpage's TeX file. However, we don't necessarily want them to be sections of the PDF version.
One solution would be to have an `\input` including the sources for all those sections directly into one section (which is followed by sections about the API and some other notes) while splitting them up into multiple sections for the man page.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**